x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -1199,26 +1196,20 @@ static void writeUnsupportedEsField(PlanStreamOutput out, UnsupportedEsField uns
 
     // -- BinaryComparison
 
-    static BinaryComparison readBinComparison(PlanStreamInput in, String name) throws IOException {
+    static EsqlBinaryComparison readBinComparison(PlanStreamInput in, String name) throws IOException {
         var source = in.readSource();
-        var operation = in.readEnum(BinaryComparisonProcessor.BinaryComparisonOperation.class);
+        EsqlBinaryComparison.BinaryComparisonOpeartion opeartion = EsqlBinaryComparison.BinaryComparisonOpeartion.readFromStream(in);
```

Comment:
This is a bit of clever magic.  I aligned the ordinals so that the ids from the old enum would still deserialize correctly into the new enum (and vice versa).  That should allow us to avoid needing a transport version change here.